src/openloco/windowmgr.cpp

```diff
@@ -136,4 +139,80 @@ namespace openloco::ui::windowmgr
         regs.esi = (uint32_t)window;
         call(0x4ca17f, regs);
     }
+
+    // 0x004C5FC8
+    void draw_single(gfx::drawpixelinfo_t* dpi, window* w, int32_t left, int32_t top, int32_t right, int32_t bottom)
+    {
+        // Copy dpi so we can crop it
+        gfx::drawpixelinfo_t copy = *dpi;
+        dpi = &copy;
+
+        // Clamp left to 0
+        int32_t overflow = left - dpi->x;
+        if (overflow > 0)
+        {
+            dpi->x += overflow;
+            dpi->width -= overflow;
+            if (dpi->width <= 0)
+                return;
+            dpi->pitch += overflow;
+            dpi->bits += overflow;
+        }
+
+        // Clamp width to right
+        overflow = dpi->x + dpi->width - right;
+        if (overflow > 0)
+        {
+            dpi->width -= overflow;
+            if (dpi->width <= 0)
+                return;
+            dpi->pitch += overflow;
+        }
+
+        // Clamp top to 0
+        overflow = top - dpi->y;
+        if (overflow > 0)
+        {
+            dpi->y += overflow;
+            dpi->height -= overflow;
+            if (dpi->height <= 0)
+                return;
+            dpi->bits += (dpi->width + dpi->pitch) * overflow;
+        }
+
+        // Clamp height to bottom
+        overflow = dpi->y + dpi->height - bottom;
+        if (overflow > 0)
+        {
+            dpi->height -= overflow;
+            if (dpi->height <= 0)
+                return;
+        }
+
+        if (/*openloco::screen_flags() & (1<<4) && */ w->type == 47)
```

Comment:
Commented

Reply:
This is your code... I don't know why it is commented.

Reply:
Because I was still working on it. I hadn't added a screen_flags method at that point